Yao - Add less-than-minimum hours logged colors on Weekly Summaries Reports page

yaow62023 opened the pull request:
# Description
Add less-than-minimum hours logged colors on Weekly Summaries Reports page (WIP Yao)
Other Links → Reports → Weekly Summaries Reports
If someone has done less than 49% of their hours, the whole Hours Logged row and hours should be red
If someone has done between 50-99% of their hours, the whole Hours Logged row and hours should be bright green
This is needed so it is clear to the Admins who these people are. Red people don’t get credited for their work for the week. Green and black people do. 

## Main changes explained:
- Add different text color for different hours logged.
- For less then 49% hours logged display as red, for 50 - 99% display as green, for 100% display as black. 

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to dashboard→ Report→ Weekly summary report
6. use a Owner account to add different Tangible Time Entry
7. test the color display correctly 

## Screenshots or videos of changes:
### less then 49% hours logged:
<img width="397" alt="Screenshot 2024-02-24 at 11 56 22 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/0b0bc923-4ef1-4d84-817d-d19dbe7f6cd4">

### 50 - 99% hours logged:
<img width="379" alt="Screenshot 2024-02-24 at 11 57 26 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/22698792-12bd-46d5-947f-ddc787e2ac2c">

### 100% hours logged:
<img width="406" alt="Screenshot 2024-02-24 at 11 55 58 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/fee2424a-f1f7-440b-9c21-47e5835740ee">

yaow62023 committed:
have some basic change on the code

yaow62023 committed:
Update FormattedReport.jsx

meetpadhiar4 reviewed (dismissed):
Hi, I reviewed your PR and hours are being color-coded correctly. Please refer the screenshots attached.

<img width="1440" alt="PR 1990 black" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/ad23d6a0-f3bb-4ffe-bced-96e370ae187c">
<img width="1440" alt="PR 1990 green" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/6c57cb60-2f0e-4aed-a995-2361f53e50d6">
<img width="1440" alt="PR 1990 red" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/d598d8a8-c6cb-475e-9d31-476d9a034466">

FyisFe commented on src/components/WeeklySummariesReport/FormattedReport.jsx:
```diff
@@ -172,6 +172,18 @@ function EmailsList({ summaries, auth }) {
   return null;
 }
 
+function getTextColorForHoursLogged(hoursLogged, promisedHours) {
+  const percentage = (hoursLogged / promisedHours) * 100;
+
+  if (percentage < 49) {
+    return 'red';
+  } else if (percentage >= 50 && percentage < 100) {
+    return 'lightgreen'; // Bright green color
+  } else {
+    return 'black'; // Default color
+  }
+}
+
```
49 <= `percentage` < 50 case is not covered, therefore it becomes black.

yaow62023 replied:
Hi, thanks for discover this bug. I have change the logic of this function. You can test it again. Thank you so much!

FyisFe reviewed (changes requested):
I tested it and found the following bug. When the percentage is 49%, it displays black

<img width="456" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59299011/87ba6e72-becd-4efb-9187-2fb5137572ee">

XiaohanMeng reviewed (changes requested):
Hi, I tested this PR and found the same issue as FyisFe: at 49% hours logged, the color turns black. Please refer to the attached video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/aed25133-9a9e-45f9-b27f-df08a89901e9

yaow62023 committed:
Update FormattedReport.jsx

fix the logic of the getTextColorForHoursLogged, and fix the bug of display numbers in two row.

yaow62023 commented:
> Hi, I tested this PR and found the same issue as FyisFe: at 49% hours logged, the color turns black. Please refer to the attached video.
> 
>  1990-result.mov

dhairya-mehta commented:
Hey , I tested and verified the following PR
<img width="1227" alt="PR 1990" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56514777/af6a32f2-1d93-4e38-99ca-81cf69c822ae">

ClemarNunes reviewed (dismissed):
I tested your PR and worked as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/134466562/853c3061-84c8-4b2d-97c4-9f934399053f

one-community commented:
Please make the green darker color code 0B6623 so it stands out more

SRamen1999 reviewed (commented):
After testing this PR all the color changes for hours logged work as intended. However I also encountered the same bug where 4.90 / 10 is not changed properly

bug
<img width="453" alt="Screen Shot 2024-04-19 at 10 11 48 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/4629c7d7-06c3-4b68-92a5-213a918b8116">


 less then 49% hours logged 
<img width="465" alt="Screen Shot 2024-04-19 at 10 15 30 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/a60e708c-9f3d-4e43-b598-c4e7ff1c0587">



50 - 99% hours logged
<img width="280" alt="Screen Shot 2024-04-19 at 10 16 19 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/671a8bb0-c7f5-4e7d-8c76-16b39d3324d8">


100% hours logged work as intended. 
<img width="597" alt="Screen Shot 2024-04-19 at 10 15 00 AM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56932664/11d6dbd3-8c73-47a1-b334-a16bfef822cc">

yaow62023 committed:
change the color of text to #0B6623

metaphor987 reviewed (dismissed):
Hi Yao, I've tested with different logged times, and the functionalities work as expected. Nice work! Below are the screenshots showing the colors for some edge cases of logged time.
![Screenshot 2024-04-21 at 1 26 23 AM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91745551/72231997-82f3-4e51-a529-d8f9fd9965a7)
![Screenshot 2024-04-21 at 1 31 49 AM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91745551/25a0cf75-9e91-4ed7-b24a-d3ad32f80894)
![Screenshot 2024-04-21 at 1 30 26 AM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91745551/d9747433-bef1-4f65-aa3f-b3116ba1da85)
![Screenshot 2024-04-21 at 1 40 13 AM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91745551/40b82cf0-6af6-434c-acbf-35e4b1b9fd99)
![Screenshot 2024-04-21 at 1 23 51 AM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91745551/03c1ccaf-1f11-4ea3-be53-9dcf6d2d4b16)

yaow62023 committed:
fix the conflict
